### DISCUSSION TITLE
Not able to install Apps

### DISCUSSION THREAD

**Discussion by chowadfrepo** [saleor/apps#214] [Q&A] [answered]
HI,

I have tried to install Search App but it's throwing the error message "_WARNING saleor.app.tasks Failed to install app. Error: The auth data given during registration request could not be used to fetch app ID._ " The same i am able to install in the my local machi but it's throwing an error in the public IP. Is there some other configuration that needs to be done when using the Public IP?

**Comment by peelar** [comment]
Hello @chowadfrepo 👋 

Can you provide step-by-step instructions on how exactly do you install the app?

**Reply by chowadfrepo** [reply]
Hi @peelar 
Installed the Apps i described in the [Apps Docs](https://github.com/saleor/apps).
1. Search started using command ` pnpm dev --filter=saleor-app-search`
2. ngrok started like `ngrok http 3000`
3. The Manifest URL is passed in the Dashboard Apps Section.

I am trying to install the Search app through the dashboard and it's Running using the ngrok.

**Reply by peelar** [reply]
Please confirm if I get you correctly. You paste the Manifest URL (with the `/api/manifest` at the end) into the "Install external app" input, and you see the error:

```
"WARNING saleor.app.tasks Failed to install app. Error: The auth data given during registration request could not be used to fetch app ID. 
```

and the app does not get installed, correct?

**Reply by krzysztofwolski** [reply]
Investigating ngrok logs could also be helpful here: https://blog.ngrok.com/posts/how-to-inspect-log-replay-traffic-with-ngrok

If the App throws the connection error, maybe a SaleorApiUrl header in the registration request is wrong?

**Reply by lkostrowski** [reply]
You can run it with `DEBUG=app-sdk:* pnpm dev --filter=saleor-app-search` and check additional debug logs

**Comment by chowadfrepo** [comment]
Hi Saleor Team,
The issue gets resolved after I mapped the domain with Saleor Core along with SSL. Looks like App installation is restricted directly through the Public IP.

The following steps I have done when App Installation Failed, I have used the APl method `file`.
Note: I have tested for Search App.

- .saleor-app-auth.json should be created automatically. (_Note: If the file is not created create it manually respective app folder_)
- Saleor Core Should be accessed through the SSL enabled Domain. ( _Note: To Importing the products into algolia won't work without SSL_),

Thanks, team for your suggestions.

**Reply by lkostrowski** [reply]
Hey, I think we need more information what do you try to do and how:
1. Are you using self hosted Saleor or Cloud? What version?
2. Are you only using `dev` app with tunnel, or you deployed version? If deployed - how?
3. What are your env variables for the app?

**Reply by chowadfrepo** [reply]
Hi @lkostrowski,

1. I am using the self hosted Saleor
2. I am using `dev` app with `ngrok` tunnel
3. The environment variable For Search App is `SECRET_KEY`

**Reply by lkostrowski** [reply]
@chowadfrepo how do you host Saleor? Does it work in docker container locally? (please check saleor-platform repository)

**Reply by chowadfrepo** [reply]
@lkostrowski I have hosted Saleor Core and Saleor Dashboard and storefront individually.

**Comment by wackyesolution** [comment]
Hello team,  i tried to clean install my saleor on vps today and deploy some apps with traditional method that worked fine for me some week ago, but now i recieve same errore error of chowadfrepo
 when i try to install this apps. (after that i passed the manifest)

Same problem on old saleor istance, and new fresh instance

**Reply by lkostrowski** [reply]
Hey, it will be hard without a reproducible demo

The error @chowadfrepo posted is caused by [saleor/app-sdk](https://github.com/saleor/saleor-app-sdk/blob/main/src/handlers/next/create-app-register-handler.ts#L185)

App, during installation uses its token and fetched its own ID from the API. This is security check.

Likely, somehow app can't reach Saleor API

please set env variable `DEBUG=app-sdk:*` and redeploy the app. You should see additional debug logs, including response from the failing request

**Comment by przlada** [comment, marked as answer]
Hi, I think that changing Saleor's domain from the default `localhost:8000` will solve your problem. Let's say you are hosting Saleor at `example.com`. You can do that with `shopDomainUpdate` mutation:
```graphql
mutation ShopDomainUpdate{
  shopDomainUpdate(input: {domain: "example.com"}){
    shop{
      domain{
        host
      }
    }
  }
}
```
or by starting the Django shell (`python manage.py migrate`) and running:
```python
from django.contrib.sites.models import Site
site = Site.objects.get_current()
site.domain = "example.com"
site.save()
```
also, remember to properly set `ENABLE_SSL` environment variable (`True` if you are using https) and to restart Saleor instances as `Site` objects are cached in memory. `ENABLE_SSL` env should be set for Saleor's API and celery worker processes/containers.

**Reply by wackyesolution** [reply]
Thank you, i solved in part. For saleor-app-checkout i fixed correctly, and in APL the domain address point correctly (finally!)
For saleor-app-cms, i used DEBUG=app-sdk:* as mentioned @lkostrowski  and this is the crash after loaded manifest correctly:







saleor-app-cms:dev: event - compiled client and server successfully in 14.5s (1263 modules)
saleor-app-cms:dev: wait  - compiling /api/manifest (client and server)...
saleor-app-cms:dev: event - compiled successfully in 2.7s (89 modules)
saleor-app-cms:dev: 2023-04-03T17:05:04.594Z app-sdk:Middleware:withBaseURL Middleware called with host: 100.XX.XX.XX:1818, protocol http
saleor-app-cms:dev: 2023-04-03T17:05:04.594Z app-sdk:Middleware:withBaseURL context.baseURL resolved to be: "http://IPXX:1818"
saleor-app-cms:dev: [sentry] If Next.js logs a warning "API resolved without sending a response", it's a false positive, which may happen when you use `withSentry` manually to wrap your routes.
saleor-app-cms:dev:               To suppress this warning, set `SENTRY_IGNORE_API_RESOLUTION_ERROR` to 1 in your env.
saleor-app-cms:dev:               To suppress the nextjs warning, use the `externalResolver` API route option (see https://nextjs.org/docs/api-routes/api-middlewares#custom-config for details).
saleor-app-cms:dev: wait  - compiling /api/register (client and server)...
saleor-app-cms:dev: event - compiled successfully in 7s (90 modules)
saleor-app-cms:dev: 2023-04-03T17:05:12.127Z app-sdk:Middleware:withSaleorDomainPresent Middleware called with domain in header: api.custom-saleordomain.com


**saleor-app-cms:dev: 2023-04-03T17:05:12.127Z app-sdk:Middleware:withAuthTokenRequired Middleware called
saleor-app-cms:dev: 2023-04-03T17:05:12.127Z app-sdk:createAppRegisterHandler Request received
saleor-app-cms:dev: 2023-04-03T17:05:12.238Z app-sdk:getAppId Could not get the app ID: SyntaxError: Unexpected token < in JSON at position 0
saleor-app-cms:dev:     at JSON.parse (<anonymous>)
saleor-app-cms:dev:     at parseJSONFromBytes (node:internal/deps/undici/undici:6498:19)
saleor-app-cms:dev:     at successSteps (node:internal/deps/undici/undici:6472:27)
saleor-app-cms:dev:     at node:internal/deps/undici/undici:1145:60
saleor-app-cms:dev:     at node:internal/process/task_queues:140:7
saleor-app-cms:dev:     at AsyncResource.runInAsyncScope (node:async_hooks:204:9)
saleor-app-cms:dev:     at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
saleor-app-cms:dev:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)**

**Reply by przlada** [reply]
I think that you need to set `ENABLE_SSL` environment variable to `True` also for the celery `worker` container (not only `api` one). Because of that, the cms app is looking for Saleor using HTTP protocol instead HTTPS.

**Reply by wackyesolution** [reply]
amazing! thank you @przlada  I confirm that ENALBLE_SSL=True on Api and Worker, worked correctly and CMS has finally been deployed! Thank you all.

Thank you also for the change domain mutation, it worked correctly and finally on the APL i can see the correct domain and not localhost!
